tests/test_lakefs_file.py

```diff
@@ -37,9 +37,13 @@ def test_lakefs_file_open_write(
 
     rpath = f"{repository}/{temp_branch}/{random_file.name}"
 
-    # try opening the remote file and writing to it
-    with fs.open(rpath, "wb") as fp:
-        fp.write(orig_text)
+    with pytest.warns(
+        UserWarning,
+        match=r"Calling `LakeFSFile\.open\(\)` in write mode results in unbuffered file uploads, because the lakeFS Python client does not support multipart uploads\. Uploading large files unbuffered can have performance implications\.",
```

Comment:
```suggestion
        match=r"Calling `LakeFSFile\.open\(\)` in write mode results in unbuffered file uploads.*",
```

In general, it is sensible to limit the `match` argument to the shortest possible piece that still conveys the message, that way you save yourself refactoring pains when you rewrite the error messages :)